knowledge-base/tooltip-in-grid.md

```diff
@@ -4,16 +4,123 @@ description: How to add Tooltips in Grid columns.
 position: 
-tags: 
+tags: telerik, blazor, grid, tooltip
 res_type: kb
 ---
 
 
 ## Description
 
-How to add Tooltips for elements in a Grid column? How to load Tooltip content on demand?
+This KB article answers the following question:
+ - How to add Tooltips for elements in a Grid column? 
+ - How to load Tooltip content on demand?
 
 
 ## Solution
 
-An example is available in the following project: [https://github.com/telerik/blazor-ui/tree/master/tooltip/in-grid](https://github.com/telerik/blazor-ui/tree/master/tooltip/in-grid).
\ No newline at end of file
+1. Use a [Column(cell) Template]({%slug grid-templates-column%}) to wrap the cell's value in a `<span>`.
+1. Attach the [`TelerikToolTip`]({%slug tooltip-overview%}) to the `<span>`.
+1. Use the [Tooltip's Template]({%slug tooltip-template%}) and its context to load Data on demand.
+
+````CSHTML
+<p>This example shows how to load content on demand for a tooltip that targets elements inside a grid. Hover over an employee name</p>
+
+<TelerikGrid Data="@GridData" 
+             Height="400px"
+             Pageable="true">
+    <GridColumns>
+        <GridColumn Field="@(nameof(BasicEmployee.Id))" Width="120px" />
+        <GridColumn Field="@(nameof(BasicEmployee.Name))" Title="Employee Name">
+            <Template>
+                @{
+                    BasicEmployee employee = context as BasicEmployee;
+                    <span data-employeeId="@employee.Id" class="tooltip-target">@employee.Name</span>
+                }
+            </Template>
+        </GridColumn>
+        <GridColumn Field="@(nameof(BasicEmployee.Team))" Title="Team" />
+    </GridColumns>
+</TelerikGrid>
+
+<TelerikTooltip TargetSelector=".tooltip-target"
+                Width="250px" Height="250px" Position="@TooltipPosition.Right">
+    <Template Context="ttipContext">
+        <h6>Employee Details</h6>
+        <hr />
+        @{
+            // in a real app, add defensive checks here or extract into a separate component
+            // the key names is the data attribute, lowercase
+            int EmployeeId = int.Parse(ttipContext.DataAttributes?["employeeid"]);
+
+            var EmployeeDetailsData = GetEmplyeeDetails(EmployeeId);
+
+            <div style="float:left; margin-left: 20px;">
+                <p><strong>Name</strong>: @EmployeeDetailsData.Name</p>
+                <p><strong>Team</strong>: @EmployeeDetailsData.Team</p>
+                <p><strong>Salary</strong>: $@EmployeeDetailsData.Salary</p>
+                <p><strong>Hire date</strong>: @EmployeeDetailsData.HireDate.ToShortDateString()</p>
+                <p><strong>Working on</strong>: @EmployeeDetailsData.ActiveProjects projects</p>
+                <p><strong>Data loaded at</strong>: @DateTime.Now.ToString("HH:mm:ss")</p>
+            </div>
+        }
+    </Template>
+</TelerikTooltip>
+
+@code {
+    private List<BasicEmployee> GridData { get; set; }
+
+    protected override async Task OnAfterRenderAsync(bool firstRender)
```

Comment:
This implementation may look strange at first. You can add a comment that this approach shows that the Tooltip will work even if the Grid is bound after the Tooltip is initialized.